tests/integration/utils.py

```diff
@@ -168,13 +209,51 @@ def __check_attributes_schema(actual_attributes, expected_attributes):
             resource_data = content['data']
             if isinstance(resource_data, list):
                 for resource in resource_data:
-                    __check_resource_schema(resource)
+                    __check_resource_schema(resource, null_value_allowed)
             else:
-                __check_resource_schema(resource_data)
+                __check_resource_schema(resource_data, null_value_allowed)
         elif status_code >= 400:
             errors_data = content['errors']
             self.assertIsInstance(errors_data, list)
             for error in errors_data:
-                __check_error_schema(error)
+                __check_error_schema(error, null_value_allowed)
```

Comment:
So, from our discussion, we need to check if there is any property we can use in openapi which allows us to specify the return fields can be null or not. If so, we can use that property to determine whether to skip the type check instead of passing `null_value_allowed` along the way. If not, we can define a list in test case level (maybe in the integration test case level) and put the fields which can/can't be null into it and use the list for the type checking.